Library/Homebrew/formulary.rb

```diff
@@ -108,7 +108,7 @@ def initialize(bottle_name)
       case bottle_name
       when %r{(https?|ftp|file)://}
         # The name of the formula is found between the last slash and the last hyphen.
-        resource = Resource.new bottle_name[%r{([^/]+)-}, 1] { url bottle_name }
+        resource = Resource.new File.basename(bottle_name)[/(.+)-/, 1] { url bottle_name }
```

Comment:
Can we split this onto 2-3 lines, it's pretty hard to follow all the parens as-is. Thanks!